ts/features/messages/components/Home/ArchiveRestoreBar.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useEffect } from "react";
+import { StyleSheet, View } from "react-native";
+import { ButtonOutline, ButtonSolid } from "@pagopa/io-app-design-system";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+import {
+  useIODispatch,
+  useIOSelector,
+  useIOStore
+} from "../../../../store/hooks";
+import {
+  areThereEntriesForShownMessageListCategorySelector,
+  isArchivingDisabledSelector,
+  isArchivingInProcessingModeSelector
+} from "../../store/reducers/archiving";
+import { useIOTabNavigation } from "../../../../navigation/params/AppParamsList";
+import { shownMessageCategorySelector } from "../../store/reducers/allPaginated";
+import I18n from "../../../../i18n";
+import {
+  resetMessageArchivingAction,
+  startProcessingMessageArchivingAction
+} from "../../store/actions/archiving";
+import { MessageListCategory } from "../../types/messageListCategory";
+import { useHardwareBackButton } from "../../../../hooks/useHardwareBackButton";
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: "row",
+    paddingHorizontal: 24,
+    paddingTop: 8
+  },
+  endButtonContainer: {
+    flex: 1,
+    marginStart: 4
+  },
+  startButtonContainer: {
+    flex: 1,
+    marginEnd: 4
+  }
+});
+
+type ArchiveRestoreCTAsProps = {
+  category: MessageListCategory;
+};
+
+export const ArchiveRestoreBar = () => {
+  const store = useIOStore();
+  const tabNavigation = useIOTabNavigation();
+
+  const isArchivingDisabled = useIOSelector(isArchivingDisabledSelector);
+  const shownCategory = useIOSelector(shownMessageCategorySelector);
+
+  const androidBackButtonCallback = useCallback(() => {
+    // Disable Android back button while processing archiving/restoring
+    const state = store.getState();
+    return isArchivingInProcessingModeSelector(state);
+  }, [store]);
+  useHardwareBackButton(androidBackButtonCallback);
+
+  useEffect(() => {
+    tabNavigation.setOptions({
+      tabBarStyle: {
+        display: isArchivingDisabled ? "flex" : "none"
+      }
+    });
```

Comment:
```suggestion
    });
     return () => {
      tabNavigation.setOptions({
        tabBarStyle: { display: "flex" }
      });
    };
```
could it be a good ieda to make sure the tabBar is always returned to its base display state this way?

Reply:
The component is never unmounted while the TabNavigator is, so such code will not run